From cf7b3a0db10cf0bcd140800ef547199703fdf16e Mon Sep 17 00:00:00 2001
From: Ashok Sidipotu <ashok.sidipotu@collabora.com>
Date: Thu, 24 Feb 2022 11:54:54 +0530
Subject: [PATCH 05/12] Revert "treewide: meson.build: get SPA_PLUGIN_DIR from
 dependency"

This reverts commit 26eb66fb5b56bb5534745a8e223bd099cf5fe4fa.
Upstream-Status: Inappropriate[meson version dependent]
---
 meson.build                          |  4 ++--
 spa/plugins/audioconvert/meson.build |  4 ++--
 spa/tests/meson.build                |  2 +-
 src/modules/meson.build              | 13 +++----------
 src/tests/meson.build                |  6 +++---
 5 files changed, 11 insertions(+), 18 deletions(-)

diff --git a/meson.build b/meson.build
index 7040e39ac..1b87bd004 100644
--- a/meson.build
+++ b/meson.build
@@ -448,8 +448,8 @@ srcdir = meson.project_source_root()
 devenv.set('PIPEWIRE_CONFIG_DIR', pipewire_dep.get_variable(internal: 'confdatadir'))
 devenv.set('PIPEWIRE_MODULE_DIR', pipewire_dep.get_variable(internal: 'moduledir'))
 
-devenv.set('SPA_PLUGIN_DIR', spa_dep.get_variable(internal: 'plugindir'))
-devenv.set('SPA_DATA_DIR', spa_dep.get_variable(internal: 'datadir'))
+  devenv.set('SPA_PLUGIN_DIR', builddir / 'spa' / 'plugins')
+  devenv.set('SPA_DATA_DIR', srcdir / 'spa' / 'plugins')
 
 devenv.set('GST_PLUGIN_PATH', builddir / 'src'/ 'gst')
 
diff --git a/spa/plugins/audioconvert/meson.build b/spa/plugins/audioconvert/meson.build
index 42332a538..3aa0f3846 100644
--- a/spa/plugins/audioconvert/meson.build
+++ b/spa/plugins/audioconvert/meson.build
@@ -140,7 +140,7 @@ foreach a : test_apps
       install : installed_tests_enabled,
       install_dir : installed_tests_execdir / 'audioconvert'),
       env : [
-        'SPA_PLUGIN_DIR=@0@'.format(spa_dep.get_variable(internal: 'plugindir')),
+        'SPA_PLUGIN_DIR=@0@/spa/plugins/'.format(meson.project_build_root()),
         ])
 
     if installed_tests_enabled
@@ -170,7 +170,7 @@ foreach a : benchmark_apps
       install : installed_tests_enabled,
       install_dir : installed_tests_execdir / 'audioconvert'),
       env : [
-        'SPA_PLUGIN_DIR=@0@'.format(spa_dep.get_variable(internal: 'plugindir')),
+        'SPA_PLUGIN_DIR=@0@/spa/plugins/'.format(meson.project_build_root()),
         ])
 
     if installed_tests_enabled
diff --git a/spa/tests/meson.build b/spa/tests/meson.build
index 0b26fdf02..d995cfaa4 100644
--- a/spa/tests/meson.build
+++ b/spa/tests/meson.build
@@ -42,7 +42,7 @@ foreach a : benchmark_apps
       install_dir : installed_tests_execdir,
     ),
     env : [
-      'SPA_PLUGIN_DIR=@0@'.format(spa_dep.get_variable(internal: 'plugindir')),
+      'SPA_PLUGIN_DIR=@0@/spa/plugins/'.format(meson.project_build_root()),
     ]
   )
 
diff --git a/src/modules/meson.build b/src/modules/meson.build
index c1225e3e0..bd7d3f711 100644
--- a/src/modules/meson.build
+++ b/src/modules/meson.build
@@ -140,12 +140,6 @@ pipewire_module_rt = shared_library('pipewire-module-rt', [ 'module-rt.c' ],
   install_rpath: modules_install_dir,
   dependencies : [dbus_dep, mathlib, dl_lib, pipewire_dep],
 )
-<<<<<<< HEAD
-# TODO: This serves as a temporary alias to prevent breaking existing setups
-#       while `module-rtkit` is being migrated to `module-rt`
-pipewire_module_rtkit = shared_library('pipewire-module-rtkit', [ 'module-rt.c' ],
-  include_directories : [configinc],
-=======
 endif
 summary({'rt': build_module_rt}, bool_yn: true, section: 'Optional Modules')
 
@@ -153,7 +147,6 @@ build_module_rtkit = dbus_dep.found()
 if build_module_rtkit
 pipewire_module_rtkit = shared_library('pipewire-module-rtkit', [ 'module-rtkit.c' ],
   include_directories : [configinc, spa_inc],
->>>>>>> Revert "treewide: meson.build: use dependency variable for SPA"
   install : true,
   install_dir : modules_install_dir,
   install_rpath: modules_install_dir,
@@ -372,9 +365,9 @@ test('pw-test-protocol-native',
     install_dir : installed_tests_execdir,
   ),
   env : [
-    'SPA_PLUGIN_DIR=@0@'.format(spa_dep.get_variable(internal: 'plugindir')),
-    'PIPEWIRE_CONFIG_DIR=@0@'.format(pipewire_dep.get_variable(internal: 'confdatadir')),
-    'PIPEWIRE_MODULE_DIR=@0@'.format(pipewire_dep.get_variable(internal: 'moduledir')),
+    'SPA_PLUGIN_DIR=@0@/spa/plugins/'.format(meson.project_build_root()),
+    'PIPEWIRE_CONFIG_DIR=@0@/src/daemon/'.format(meson.project_build_root()),
+    'PIPEWIRE_MODULE_DIR=@0@/src/modules/'.format(meson.project_build_root())
   ]
 )
 
diff --git a/src/tests/meson.build b/src/tests/meson.build
index 3e8d05012..9a595d431 100644
--- a/src/tests/meson.build
+++ b/src/tests/meson.build
@@ -13,9 +13,9 @@ foreach a : test_apps
       install : installed_tests_enabled,
       install_dir : installed_tests_execdir),
     env : [
-      'SPA_PLUGIN_DIR=@0@'.format(spa_dep.get_variable(internal: 'plugindir')),
-      'PIPEWIRE_CONFIG_DIR=@0@'.format(pipewire_dep.get_variable(internal: 'confdatadir')),
-      'PIPEWIRE_MODULE_DIR=@0@'.format(pipewire_dep.get_variable(internal: 'moduledir')),
+      'SPA_PLUGIN_DIR=@0@/spa/plugins/'.format(meson.project_build_root()),
+      'PIPEWIRE_CONFIG_DIR=@0@/src/daemon/'.format(meson.project_build_root()),
+      'PIPEWIRE_MODULE_DIR=@0@/src/modules/'.format(meson.project_build_root())
       ])
 
   if installed_tests_enabled
-- 
2.35.1

